pkg/util/deployment/deployment.go

```diff
@@ -48,6 +48,39 @@ const (
 	RollbackDone              = "DeploymentRollback"
 )
 
+// GetAllReplicaSets list all RSes of a given deployment atomically
+func GetAllReplicaSets(deployment *extensions.Deployment, c clientset.Interface) (allOldRSes []*extensions.ReplicaSet, newRS *extensions.ReplicaSet, err error) {
```

Comment:
The deployment controller adds labels matching the selector to the ReplicaSets, yes.